airbyte-integrations/connectors/source-trello/source_trello/source.py

```diff
@@ -0,0 +1,227 @@
+    API Docs: https://developer.atlassian.com/cloud/trello/rest/api-group-boards/#api-boards-id-checklists-get
+    Endpoint: https://api.trello.com/1/boards/<id>/checklists
+    """
+
+    parent_stream_class = Boards
+    extra_params = {"fields": "all", "checkItem_fields": "all"}
+
+    def path(self, stream_slice: Mapping[str, Any] = None, **kwargs) -> str:
+        return f"boards/{stream_slice['id']}/checklists"
+
+
+class Lists(ChildStreamMixin, TrelloStream):
+    """Return list of all lists of a boards.
+    API Docs: https://developer.atlassian.com/cloud/trello/rest/api-group-boards/#api-boards-id-lists-get
+    Endpoint: https://api.trello.com/1/boards/<id>/lists
+    """
+
+    parent_stream_class = Boards
+
+    def path(self, stream_slice: Mapping[str, Any] = None, **kwargs) -> str:
+        return f"boards/{stream_slice['id']}/lists"
+
+
+class Users(ChildStreamMixin, TrelloStream):
+    """Return list of all members of a boards.
+    API Docs: https://developer.atlassian.com/cloud/trello/rest/api-group-boards/#api-boards-id-members-get
+    Endpoint: https://api.trello.com/1/boards/<id>/members
+    """
+
+    parent_stream_class = Boards
+
+    def path(self, stream_slice: Mapping[str, Any] = None, **kwargs) -> str:
+        return f"boards/{stream_slice['id']}/members"
+
+
+class Actions(ChildStreamMixin, IncrementalTrelloStream):
+    """Return list of all actions of a boards.
+    API Docs: https://developer.atlassian.com/cloud/trello/rest/api-group-boards/#api-boards-boardid-actions-get
+    Endpoint: https://api.trello.com/1/boards/<id>/actions
+    """
+
+    parent_stream_class = Boards
+    limit = 1000
+
+    def path(self, stream_slice: Mapping[str, Any] = None, **kwargs) -> str:
+        return f"boards/{stream_slice['id']}/actions"
+
+
+class TrelloAuthenticator(HttpAuthenticator):
+    def __init__(
+        self,
+        token: str,
+        key: str,
+        auth_header: str = "Authorization",
+        key_header: str = "oauth_consumer_key",
+        token_header: str = "oauth_token",
+    ):
+        self.auth_header = auth_header
+        self.key_header = key_header
+        self.token_header = token_header
+        self._key = key
+        self._token = token
+
+    def get_auth_header(self) -> Mapping[str, Any]:
+        return {self.auth_header: f'OAuth {self.key_header}="{self._key}", {self.token_header}="{self._token}"'}
+
+
+class SourceTrello(AbstractSource):
+    """
+    Source Trello fetch date from web-based, Kanban-style, list-making application.
+    """
+
+    def check_connection(self, logger, config) -> Tuple[bool, any]:
+        """
+        Testing connection availability for the connector by granting the credentials.
+        """
+
+        try:
+            url = f"{TrelloStream.url_base}members/me"
+
+            authenticator = TrelloAuthenticator(token=config["token"], key=config["key"])
+
+            session = requests.get(url, headers=authenticator.get_auth_header())
```

Comment:
Can we check possible permissions and streams here?

Reply:
Those code do already do what you ask.

Reply:
do default settings provide access for all streams?

Reply:
Yes.